app/controllers/deploys_controller.rb

```diff
@@ -98,6 +98,7 @@ def changeset
   def destroy
     if @deploy.can_be_stopped_by?(current_user)
       @deploy.stop!
+      SseRailsEngine.send_event('deploys', { type: 'finish' })
```

Comment:
Aye, the way the DeployService and Deploy classes are written/used makes it awkward to slot it in... I'll have a look.